### ISSUE TITLE
Very long string issue.

### ISSUE THREAD

**Issue by josecvega** [kach/nearley#502]
I would like to start by sayingI've been using this library for almost a year to parsed our made up language and it's worked without any issues.

Just recently we've maintained our existing grammar but we support strings up to 128,000 characters and after some initial testing we are experiencing Javascript out of memory issues.

My issue is more of a question. Has anyone had similar issues or could it be something related to the grammar?

**Comment by wouterbeek** [comment]
Yes, I saw this as well.  I have had OOM issues with ~30K character strings.

**Comment by Henddher** [comment]
> I am running into it too (with a smaller input file 12K). 
> Increasing heap space hasn't helped "yet" (`node --max-old-space-size`).
> My very wild guess is that the heap is fragmented.
> Next step:
> https://nodejs.org/en/learn/diagnostics/memory/using-gc-traces

The grammar was ambiguous.
